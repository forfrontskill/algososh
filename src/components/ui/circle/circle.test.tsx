import React from 'react';
import { render, screen } from '@testing-library/react';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';
import renderer from 'react-test-renderer';
describe('Test Circle component', () => {

    it('Button rendering', () => {
        const tree = renderer
          .create(<Circle/>)
          .toJSON();
          expect(tree).toMatchSnapshot();
      });

    it('Circle creation', ()=>{
        const context = render(<Circle />);

        const circle = screen.getByTestId('circle');
    
        expect(circle).toBeInTheDocument();
    });

    it('Circle without text', ()=>{
        const context = render(<Circle />);

        const circleText = screen.getByTestId('circle-text').textContent;
    
        expect(circleText).toBe('');
    });

    it('Circle with text', ()=>{
        const text = 'TestText';
        const context = render(<Circle letter={text}/>);

        const circleText = screen.getByTestId('circle-text').textContent;
    
        expect(circleText).toBe(text);
    });

    it('Circle head', ()=>{
        const text = 'TestText';
        const context = render(<Circle head={text}/>);

        const circleHeadText = screen.getByTestId('circle-head').textContent;
    
        expect(circleHeadText).toBe(text);
    });

    it('Circle head element', ()=>{
        const element = (<div data-testid="element-test"></div>)
        const context = render(<Circle head={element}/>);

        const elementHead = screen.getByTestId('element-test');
    
        expect(elementHead).toBeInTheDocument();
    });

    it('Circle tail', ()=>{
        const text = 'TestText';
        const context = render(<Circle tail={text}/>);

        const circleTailText = screen.getByTestId('circle-tail').textContent;
    
        expect(circleTailText).toBe(text);
    });

    it('Circle tail element', ()=>{
        const element = (<div data-testid="element-test"></div>)
        const context = render(<Circle tail={element}/>);

        const elementTail = screen.getByTestId('element-test');
    
        expect(elementTail).toBeInTheDocument();
    });

    it('Circle index', ()=>{
        const index = 1;
        const context = render(<Circle index={index}/>);

        const circleIndex = screen.getByTestId('circle-index').textContent;
    
        expect(circleIndex).toBe(index.toString());
    });

    it('Circle smile', ()=>{
        const context = render(<Circle isSmall={true}/>);

        const isSmall = screen.getByTestId('circle-main').classList.contains('small');
        expect(isSmall).toBe(true);
    });

    it('Circle default', ()=>{
        const context = render(<Circle state={ElementStates.Default}/>);

        const isSmall = screen.getByTestId('circle-main').classList.contains('default');
        expect(isSmall).toBe(true);
    });

    it('Circle default', ()=>{
        const context = render(<Circle state={ElementStates.Changing}/>);

        const isSmall = screen.getByTestId('circle-main').classList.contains('changing');
        expect(isSmall).toBe(true);
    });

    it('Circle default', ()=>{
        const context = render(<Circle state={ElementStates.Modified}/>);

        const isSmall = screen.getByTestId('circle-main').classList.contains('modified');
        expect(isSmall).toBe(true);
    });
});