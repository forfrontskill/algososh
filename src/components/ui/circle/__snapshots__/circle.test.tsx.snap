// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Circle component Button rendering 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle-main"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-text"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;
